File: android/core/network/bin/main/com/aibook/android/core/network/opds/OpdsModels.kt
package com.aibook.android.core.network.opds

data class OpdsFeed(
    val title: String,
    val entries: List<OpdsEntry>,
    val nextLink: OpdsLink? = null
)

data class OpdsEntry(
    val title: String,
    val identifier: String? = null,
    val author: String? = null,
    val summary: String? = null,
    val acquisitionLink: OpdsLink? = null,
    val alternateLink: OpdsLink? = null,
    val coverLink: OpdsLink? = null,
    val categories: List<String> = emptyList(),
    val modifiedAt: String? = null
)

data class OpdsLink(
    val href: String,
    val type: String? = null,
    val rel: String? = null
)

data class OpdsConnection(
    val id: String,
    val name: String,
    val baseUrl: String,
    val username: String? = null,
    val password: String? = null,
    val enabled: Boolean = true,
    val lastSyncedAt: Long? = null,
    val bookCount: Int = 0,
    val syncState: OpdsSyncState = OpdsSyncState.IDLE,
    val lastErrorMessage: String? = null,
    val syncMode: OpdsSyncMode = OpdsSyncMode.FULL
)

enum class OpdsSyncMode { FULL, INCREMENTAL }

enum class OpdsSyncState {
    IDLE,
    SYNCING,
    SUCCESS,
    FAILED
}

File: ios/AiBook/Core/Model/OpdsModels.swift
import Foundation

// MARK: - OpdsFeed（与安卓 OpdsFeed 对齐）

struct OpdsFeed: Sendable {
    let title: String?
    let entries: [OpdsEntry]
}

// MARK: - OpdsEntry（与安卓 OpdsEntry 对齐）

struct OpdsEntry: Sendable {
    let title: String
    let author: String?
    let summary: String?
    let acquisitionLink: OpdsLink?
    let alternateLink: OpdsLink?
    let coverLink: OpdsLink?
}

// MARK: - OpdsLink（与安卓 OpdsLink 对齐）

struct OpdsLink: Sendable {
    let href: String
    let type: String?
    let rel: String?
}

// MARK: - OpdsConnection（与安卓 OpdsConnection 对齐）

struct OpdsConnection: Identifiable, Sendable {
    let id: String
    var name: String
    var baseUrl: String
    var username: String?
    var password: String?
    var enabled: Bool
    var lastSyncedAt: Date?
    var bookCount: Int
    var syncState: OpdsSyncState
    var lastErrorMessage: String?
}

// MARK: - OpdsSyncState（与安卓枚举对齐）

enum OpdsSyncState: String, Codable, Sendable {
    case idle
    case syncing
    case success
    case failed
}
